app/controllers/vm_common.rb

```diff
@@ -1186,11 +1214,28 @@ def open_parent_nodes(record)
     add_nodes
   end
 
+  # if node is VM or Template and hide_vms is true - select parent node in explorer tree but show info of Vm/Template
+  def resolve_node_info(id)
+    nodetype, id = id.split("-")
+
+    if hide_vms && (nodetype == 'v' || nodetype == 't')
+      self.x_node = parent_folder_id(VmOrTemplate.find(id))
+      @vm = VmOrTemplate.find(id)
```

Comment:
find is being called twice here for the same VM...you can probably swap these two lines and reuse the `@vm` variable to not incur a second query.